controller/state.go

```diff
@@ -136,10 +136,8 @@ func (m *appStateManager) getRepoObjs(app *v1alpha1.Application, source v1alpha1
 	if err != nil {
 		return nil, nil, nil, err
 	}
-
-	targetObjs, hooks, nil := unmarshalManifests(manifestInfo.Manifests)
-	return targetObjs, hooks, manifestInfo, nil
-
+	targetObjs, hooks, err := unmarshalManifests(manifestInfo.Manifests)
```

Comment:
Why not the following? Seems more explicit and fits the Go error pattern.
```suggestion
	targetObjs, hooks, err := unmarshalManifests(manifestInfo.Manifests)
	if err != nil {
		return nil, nil, nil, err
	}
```